Project-2/README.md

```diff
@@ -100,15 +98,15 @@ D: {(Severed_Hand, Maria),(Maria,Severed_Hand)}
+- Languages in logic are like special clubs, and these clubs can be used to make new clubs about whatever we're interested in. Sometimes, us logicians (logical people) want to know if something new can be part of a club or not. We can test it out by treating what we're interested in like a new member, and seeing if it gets along with the other members. If we can count the number of steps it takes before we can tell if everybody gets along or not, our club has the finite model property.
 - The finite model property is important because it gives us a helpful routine that makes figuring out these clubs a lot simpler. Sometimes all we need is three members to tell us they can't handle a new member. 
-- Now, sometimes we don't want to limit the members of these clubs. Then it's more about making sure there can be space for everybody, even if that takes more work. When we want to make clubs like these, we don't need the finite model property.
+- Now, sometimes we don't want to limit the members of these clubs. Then it's more about making sure there can be space for everybody, even if that takes more work. When we're thinking about clubs like this, we don't need the finite model property so much.
 
 8. Following up on the preceding , explain the _tree model property_. Be sure to provide a simple example and explain when the property might be important, and when it is not so important. 
-- One way us logicians think about what we're interested is by picturing it like it's the trunk of a tree. 
+- One way logicians (logical people) think about what we're interested is by picturing it like it's the trunk of a tree. 
 - A tree has lots of different branches that go in different directions. These branches _extend away_ from the trunk, but they're a part of the whole tree just like every other part is. The ends of these branches grow into more specific parts of what we want to think about, like fruits or leaves. These are connected to the whole rest of the tree by the branches. 
 - As the tree grows, us logicians get to collect what we're interested in and look at it all together. We can add our own branches, more fruits, or even ask computers to lend a hand. Looking at a tree can make complicated problems easier to figure out.
-- Sometimes, though, if we tried to make a tree out of what we're looking at, it wouldn't help us that much. Maybe the tree it would make is too small, or too weak. Then we'll find other ways of putting the parts together instead of trying to grow one. 
+- Sometimes, though, if we tried to make a tree out of what we're looking at, it wouldn't help us that much. Then we'll find other ways of putting the parts together instead of trying to grow one. 
```

Comment:
Hi @oliviahobai , here are my comments

Q1. (a) I prefer the second formulation and delete the existential quantifier, but this is of course controversial so I will leave you to determine that. My justification is:

Suppose you are right that PP≡∃(P ⊓ ¬P¯) [I recommend to change your expression of negation sign or subscript like this]. Now we wanna say that a is a proper part of b, then we can say: (a,b):PP. Given your definition, we have: (a,b): ∃(P ⊓ ¬P¯). You are unable to explain what is bounded by this “∃”. After all, “a” and “b” are both individual constants but not variables. Then, what else variable can be bounded by it? This is why I delete “∃”.

(d) According to your definition, a overlaps b iff a is a part of some x, and x has b as its part. In this case, a and b may be two disjoint parts of x, so your definition is wrong. Instead, you should write it as: O ≡ ∃P¯.(∃P). 

“Could ∃(P⊓P^_) also work for the above?” No! If it works, then a overlaps b iff a is a part of b and b is a part of a (and thus a = b). But a can overlap b without being equal to it.

(e) “D:= 0^_” is wrong because D is the negation of O, so you wanna write: D≡ ¬ O.

Q2. (1) “T Box: inclusive Terms.” – Yep, T-box includes all 5 formulas in Q1, but you should supplement it with one or two extra formulas like PP ⊑P.

(2) Note the distinction between individual names and elements (in domain). When you say “Instances: {Hand, Severed_Hand, Handless_Darcy, Darcy_2Hands}, you should assign these names to individuals a, b, c, d if they are “names”, otherwise you should call this set as the domain △ of your model.

(3) Concepts: {Limb, Person} – It seems that you don’t need any concept here. You can directly say two individuals are in a relation without mentioning any concept.

(4) “Relations: {P, PP, P¯, PP¯, 0, D}” – You should tell us the extensions of these relations/roles. For example, you can say something like “PP={(a,b), (b,c), …}”

Q3. (a) My new attempt: ∃R.(∀R.(∀R¯)). This means: Fixing x (restricted by an impicit universal quantifer), all x is R-related to some y such that y is R-related to all z such that z is R¯-related to all x (or, all x is R-related to z).

(b) My new attempt: ∃R¯.(∃R.(∃R¯)). This means: Fixing y (restricted by the only universal quantifer), y is R-inverse-related to some x, which is then related to some z, which is then R-inverse-related to all y.

(c) “ALC does not include conditionals or biconditionals in its syntax.” -- Really? PP ⊑P is conditional in ALC, and P≡P is biconditional/equivalent formula in ALC. 

The formula (c) can be translated as: ∀R.∃R.∀R.A

(d) (∀R.A) ⊓ (∃R.B) – This is very straightforward, please read Baader’s book, pp. 45 for further explanations.

Note: Here we are not required to do translations in FOL, but to translate formulas in first-order logic to corresponding formulas in description logic.

Q4. The key difference between ALC and ALCN is, ALCN have numerical expressions. Keep in mind that a numerical expression is used to say how many R-fillers an individual has. So you can give any a formula including numerical restriction, say, (≤1 r), and then find two models, one of which includes a point that has no R-filler, and another of which includes a point that has at most one R-filler. In this way, you can use (≤1 r) to differ between those two models.

Q5. “I feel like this could be done by contrasting (bisimulating?) two solitary sets of relations.” – **A genius idea!** You can draw a line with two end points in model 1, and then draw it and an isolated copy of it –that is, two isolated lines-- in model 2. These two models are bisimilar. 

Q6. (b) ∃r-.C means that x has a r-predecessor which is an instance of C; and ∀r-.C means x's all r-predecessors are instances of C. 

Predecessor vs. Successor (filler): If we wanna say r(a, b), then a is before b, or a is predecessor of b, so a is r-related to b (or equivalently, b is r-inversed-related to a). But if we wanna say r(b,a), then a is after b, or a is successor/filler of b, so a is r-inversed-related to b (or equivalently, b is r-related to a).

(c) “The first expression specifies there are fewer than or equal to n instances of the relation 'r'” 

Note: Only Concepts have instances, but Roles/Relations have no instance. So you cannot say “n instances of the relation r”. \

Instead, you should say, “All x have fewer than or equal to n R-fillers, or All x are related to fewer than or equal to n individuals via the relation R”.

Similarly, the second one says, “All x are R-related to fewer than or equal to n instances of C.”

(d) “There could be multiple instances of C being 'r'd'.” 

Instead, say: Some instance of C is R-related to all x.

“∃r-.{a}” means: The individual a is R-related to all x, or all x are R-inverse-related to the individual a. 

Q7-8. Sounds elegant. A bit notes:

Finite model property (FMP): Any satisfiable concept can be satisfied by a finite model.

Tree model property (TMP): Any satisfiable concept can be satisfied by a tree model.

When I read Baadar’s book, I find the following points:
(1) Finite model property (FMP) can be used to design a decidable algorithm for the satisfiability of its concepts.
(2) TMP is crucial to tableau-based algorithms for the consistency check of knowledge base and applied ontology. Most of applied ontologies, especially those BFO-based ones, have TMP, allowing that their consistency can be checked by tableau reasonings.

Q9.(1) Parthood is not symmetric: New York is a part of US, but US is not a part New York.
(2) The key step is if P is a sub-property of O, and c is a part of a, then c overlaps a.